src/modules/layoutUiRules/rulesHelper.js

```diff
@@ -38,3 +38,25 @@ export const getAxisPerLockedDimByVisType = (visType, dimensionId) =>
 
 export const getAllLockedDimIdsByVisType = visType =>
     Object.keys(getLockedDimsByVisType(visType))
+
+export const isDimensionLocked = (visType, dimensionId) =>
+    getAllLockedDimIdsByVisType(visType).find(id => id === dimensionId)
+
+export const isAxisFull = (visType, axisId, axisDimensionsCount) =>
+    axisDimensionsCount === getAxisMaxNumberOfDimsByVisType(visType, axisId)
+
+export const canDimensionBeAddedToAxis = (visType, layout, axisId) => {
```

Comment:
why pass the whole `layout` when only `layout[axisId]` is used?

Reply:
Fixed